ibm/resource_ibm_pi_capture.go

```diff
@@ -6,198 +6,219 @@ package ibm
-		CloudStorageAccessKey: "",
-		CloudStorageImagePath: cloudstorageImagePath,
-		//CloudStorageRegion:   ptrToString(cloudstorageregion),
-		CloudStorageSecretKey: "",
+	if v, ok := d.GetOk(helpers.PIInstanceCaptureVolumeIds); ok {
+		volids := expandStringList((v.(*schema.Set)).List())
+		if len(volids) > 0 {
+			captureBody.CaptureVolumeIds = volids
+		}
 	}
 
-	err = client.CaptureInstanceToImageCatalog(name, body)
+	captureResponse, err := client.CaptureInstanceToImageCatalogV2(name, captureBody)
 
 	if err != nil {
-		return errors.New("the capture cannot be performed")
+		return diag.FromErr(err)
 	}
 
-	// If this is an image catalog then we need to check what the status is
-
-	imageClient := st.NewIBMPIImageClient(context.Background(), sess, powerinstanceid)
-	imagedata, err := imageClient.Get(d.Get(helpers.PIInstanceCaptureName).(string))
-
+	jobClient := st.NewIBMPIJobClient(ctx, sess, cloudInstanceID)
+	_, err = waitForIBMPIJobCompleted(ctx, jobClient, *captureResponse.ID, d.Timeout(schema.TimeoutCreate))
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-	log.Printf("Printing the data %s - %s", *imagedata.ImageID, imagedata.State)
-
-	_, err = isWaitForImageCaptureAvailable(client, *imagedata.ImageID, powerinstanceid, d.Timeout(schema.TimeoutCreate))
-
-	//_, err = isWaitForIBMPIVolumeAvailable(client, d.Id(), powerinstanceid, d.Timeout(schema.TimeoutCreate))
-	//if err != nil {
-	//	return err
-	//}
-	return nil
-	//return resourceIBMPIVolumeAttachRead(d, meta)
-
+	d.SetId(fmt.Sprintf("%s/%s/%s", cloudInstanceID, capturename, capturedestination))
+	return resourceIBMPICaptureRead(ctx, d, meta)
 }
 
-func resourceIBMPICaptureRead(d *schema.ResourceData, meta interface{}) error {
-
-	return nil
-}
-
-func resourceIBMPICaptureUpdate(d *schema.ResourceData, meta interface{}) error {
-
-	sess, _ := meta.(ClientSession).IBMPISession()
-	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
-	client := st.NewIBMPIVolumeClient(context.Background(), sess, powerinstanceid)
-
-	name := ""
-	if d.HasChange(helpers.PIVolumeAttachName) {
-		name = d.Get(helpers.PIVolumeAttachName).(string)
-	}
-
-	size := float64(d.Get(helpers.PIVolumeSize).(float64))
-	shareable := bool(d.Get(helpers.PIVolumeShareable).(bool))
-
-	volrequest, err := client.UpdateVolume(d.Id(), &models.UpdateVolume{Name: &name, Size: size, Shareable: &shareable})
+func resourceIBMPICaptureRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(ClientSession).IBMPISession()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-
-	_, err = isWaitForIBMPIVolumeAvailable(context.Background(), client, *volrequest.VolumeID, d.Timeout(schema.TimeoutCreate))
+	cloudInstanceID, captureID, err := splitID(d.Id())
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-
-	return resourceIBMPICaptureRead(d, meta)
+	capturedestination := d.Get(helpers.PIInstanceCaptureDestination).(string)
```

Comment:
As mentioned earlier get this value from Id parts above instead of d.Get.

Reply:
Done